### ISSUE TITLE
Feature: more flexibility for image sizes

### ISSUE THREAD

**Issue by nicksarro** [WPprodigy/woocommerce-external-product-embed#9]
Not sure if it's just me, but none of these are working for me? - https://gist.github.com/WPprodigy/96dbaeab8820cc85af42 

Where am I supposed to add these codes?

**Comment by nicksarro** [comment]
I added it into my theme functions.php, but maybe I'm wrong in doing this?

**Comment by nicksarro** [comment]
Also if someone could help me with how I might change the size of the product titles? 
I'm really not good at this stuff. I Really Appreciate the help!

**Comment by BurlesonBrad** [comment]
@nicksarro , I'd put the "extra" somewhere within @WPprodigy plugin folder. Not the theme.

You've connected the API's and all that correct?

**Comment by nicksarro** [comment]
@BurlesonBrad The products show perfect, I just want to make all the images one cropped size. I tried with some basic css, but it get funky. So I thought adding this extra will help. I'll try your advice and see where I get! Thank you.

**Comment by nicksarro** [comment]
@BurlesonBrad  - Just created a php file inside the plugin, but still no change in image size. Any other suggestions?

**Comment by WPprodigy** [comment]
Hey @nicksarro,

Oh, the custom code in that file won't help you change the image size. Here is a filter for where the image is retrieved: https://github.com/WPprodigy/woocommerce-external-product-embed/blob/master/classes/class-woocommerce-external-product-embed-transients.php#L143

And here are the available image properties from the api right now: http://woothemes.github.io/woocommerce-rest-api-docs/#images-properties

So "src" is being pulled from here in WooCommerce core: https://github.com/woothemes/woocommerce/blob/master/includes/api/legacy/v3/class-wc-api-products.php#L2209. As you can see, it is grabbing the "full" version. 

___

tl;dr, instead of pulling the 'src' in my plugin, you could pull the attachment ID (https://github.com/woothemes/woocommerce/blob/master/includes/api/legacy/v3/class-wc-api-products.php#L2216). And then run your own `wp_get_attachment_image_src()` function to retrieve your preferred image size: https://developer.wordpress.org/reference/functions/wp_get_attachment_image_src/

**Comment by nicksarro** [comment]
@WPprodigy Thanks for the reply I think I should be able to follow your direction! There isn't an easier fix for just cropping them is there? I've been playing around with it and almost there, but image look stretched and not cropped. Thank you for your help!

**Comment by WPprodigy** [comment]
There isn't an easier way at the moment, but after researching this, I can make it easier in the extension to do this. I will probably switch to pulling the attachment ID, and make the image size filterable.

To make this work right now, you will need to manually edit the core plugin files and pass $product along in this filter: https://github.com/WPprodigy/woocommerce-external-product-embed/blob/master/classes/class-woocommerce-external-product-embed-transients.php#L143

**Comment by WPprodigy** [comment]
Closing this out, as the API still doesn't have much flexibility for image options. The best I can do still is grab the source: http://woocommerce.github.io/woocommerce-rest-api-docs/#product-properties
